docs/resources/grants.md

```diff
@@ -3,38 +3,59 @@ subcategory: "Unity Catalog"
-- `ALL_PRIVILEGES`: Applicable object types: All object types. Allows a user to grant or revoke all privileges applicable to the securable without explicitly specifying them. This expands to all available privileges at the time of the grant.
-
-In Unity Catalog, privileges are not inherited on child securable objects. For example, if you grant the `CREATE` privilege on a catalog to a user, the user does not automatically have the `CREATE` privilege on all schemas in the catalog.
+Securable objects are hierarchical and privileges are inherited downward. The highest level object that privileges are inherited from is the catalog. This means that granting a privilege on a catalog or schema automatically grants the privilege to all current and future objects within the catalog or schema. Privileges that are granted on a metastore are not inherited.
 
 Every `databricks_grants` resource must have exactly one securable identifier and one or more `grant` blocks with the following arguments:
 
 - `principal` - User or group name.
 - `privileges` - One or more privileges that are specific to a securable type.
 
+The securable objects are:
+
+- `METATORE`: The top-level container for metadata. Each metastore exposes a three-level namespace (`catalog`.`schema`.`table`) that organizes your data.
+- `CATALOG`: The first layer of the object hierarchy, used to organize your data assets.
+- `SCHEMA`: Also known as databases, schemas are the second layer of the object hierarchy and contain tables and views.
+- `TABLE`: The lowest level in the object hierarchy, tables can be  _external_ (stored in external locations in your cloud storage of choice) or _managed_ tables (stored in a storage container in your cloud storage that you create expressly for UC).
+- `VIEW`: A read-only object created from one or more tables that is contained within a schema.
+- `EXTERNAL LOCATION`: An object that contains a reference to a storage credential and a cloud storage path that is contained within a metatore.
+- `STORAGE CREDENTIAL`: An object that encapsulates a long-term cloud credential that provides access to cloud storage that is contained within a metatore.
+- `FUNCTION`: A user defined function that is contained within a schema.
+- `SHARE`: A logical grouping for the tables you intend to share using Delta Sharing. A share is contained within a metastore.
+- `RECIPIENT`: An object that identifies an organization or group of users that can have data shared with them using Delta Sharing. These objects are contained within a metastore.
+- `PROVIDER`: An object that represents an organization that has made data available for sharing using Delta Sharing. These objects are contained within a metastore.
```

Comment:
we don't yet have mapping in `databricks_grants` for delta sharing. can you remove these three from this PR and add them in a separate one, along with implementation for sharing permissions?